### ISSUE TITLE
[vagrant-build-images] Failed to build FreeBSD 12 cloud image

### ISSUE THREAD

**Issue by kgodzwox** [spdk/spdk#2464]
# CI Intermittent Failure

Failed to build FreeBSD 12 cloud image.

## Execution failed at

```
[2022-04-13T04:26:03.919Z] 2022-04-13T06:26:03+02:00:     freebsd12: [197/233] Linking static target drivers/libtmp_rte_bus_pci.a

[2022-04-13T04:26:03.919Z] 2022-04-13T06:26:03+02:00:     freebsd12: [198/233] Generating drivers/rte_bus_vdev.pmd.c with a custom command

[2022-04-13T04:26:03.919Z] 2022-04-13T06:26:03+02:00:     freebsd12: FAILED: drivers/rte_bus_vdev.pmd.c

[2022-04-13T04:26:03.919Z] 2022-04-13T06:26:03+02:00:     freebsd12: /usr/local/bin/python3.8 ../buildtools/gen-pmdinfo-cfile.py /git_repos/spdk_abi_latest/dpdk/build-tmp/buildtools ar /git_repos/spdk_abi_latest/dpdk/build-tmp/drivers/libtmp_rte_bus_vdev.a drivers/rte_bus_vdev.pmd.c /usr/local/bin/python3.8 ../buildtools/pmdinfogen.py elf

[2022-04-13T04:26:03.919Z] 2022-04-13T06:26:03+02:00:     freebsd12: ar: fatal: Unrecognized archive format: Inappropriate file type or format

[2022-04-13T04:26:03.919Z] 2022-04-13T06:26:03+02:00:     freebsd12: Traceback (most recent call last):

[2022-04-13T04:26:03.919Z] 2022-04-13T06:26:03+02:00:     freebsd12:   File "../buildtools/gen-pmdinfo-cfile.py", line 13, in <module>

[2022-04-13T04:26:03.919Z] 2022-04-13T06:26:03+02:00:     freebsd12:     for name in subprocess.run([ar, "t", archive], stdout=subprocess.PIPE,

[2022-04-13T04:26:03.919Z] 2022-04-13T06:26:03+02:00:     freebsd12:   File "/usr/local/lib/python3.8/subprocess.py", line 516, in run

[2022-04-13T04:26:03.919Z] 2022-04-13T06:26:03+02:00:     freebsd12:     raise CalledProcessError(retcode, process.args,

[2022-04-13T04:26:03.919Z] 2022-04-13T06:26:03+02:00:     freebsd12: subprocess.CalledProcessError: Command '['ar', 't', '/git_repos/spdk_abi_latest/dpdk/build-tmp/drivers/libtmp_rte_bus_vdev.a']' returned non-zero exit status 1.

[2022-04-13T04:26:03.919Z] 2022-04-13T06:26:03+02:00:     freebsd12: [199/233] Generating drivers/rte_bus_pci.pmd.c with a custom command

[2022-04-13T04:26:03.919Z] 2022-04-13T06:26:03+02:00:     freebsd12: FAILED: drivers/rte_bus_pci.pmd.c

[2022-04-13T04:26:03.919Z] 2022-04-13T06:26:03+02:00:     freebsd12: /usr/local/bin/python3.8 ../buildtools/gen-pmdinfo-cfile.py /git_repos/spdk_abi_latest/dpdk/build-tmp/buildtools ar /git_repos/spdk_abi_latest/dpdk/build-tmp/drivers/libtmp_rte_bus_pci.a drivers/rte_bus_pci.pmd.c /usr/local/bin/python3.8 ../buildtools/pmdinfogen.py elf

[2022-04-13T04:26:03.919Z] 2022-04-13T06:26:03+02:00:     freebsd12: ar: fatal: Unrecognized archive format: Inappropriate file type or format

[2022-04-13T04:26:03.919Z] 2022-04-13T06:26:03+02:00:     freebsd12: Traceback (most recent call last):

[2022-04-13T04:26:03.919Z] 2022-04-13T06:26:03+02:00:     freebsd12:   File "../buildtools/gen-pmdinfo-cfile.py", line 13, in <module>

[2022-04-13T04:26:03.919Z] 2022-04-13T06:26:03+02:00:     freebsd12:     for name in subprocess.run([ar, "t", archive], stdout=subprocess.PIPE,

[2022-04-13T04:26:03.919Z] 2022-04-13T06:26:03+02:00:     freebsd12:   File "/usr/local/lib/python3.8/subprocess.py", line 516, in run

[2022-04-13T04:26:03.919Z] 2022-04-13T06:26:03+02:00:     freebsd12:     raise CalledProcessError(retcode, process.args,

[2022-04-13T04:26:03.919Z] 2022-04-13T06:26:03+02:00:     freebsd12: subprocess.CalledProcessError: Command '['ar', 't', '/git_repos/spdk_abi_latest/dpdk/build-tmp/drivers/libtmp_rte_bus_pci.a']' returned non-zero exit status 1.

[2022-04-13T04:26:03.919Z] 2022-04-13T06:26:03+02:00:     freebsd12: [200/233] Generating symbol file lib/librte_mbuf.so.22.0.p/librte_mbuf.so.22.0.symbols

[2022-04-13T04:26:04.487Z] 2022-04-13T06:26:04+02:00:     freebsd12: [201/233] Compiling C object lib/librte_reorder.a.p/reorder_rte_reorder.c.o

[2022-04-13T04:26:04.487Z] 2022-04-13T06:26:04+02:00:     freebsd12: [202/233] Compiling C object lib/librte_cryptodev.a.p/cryptodev_rte_cryptodev.c.o

[2022-04-13T04:26:04.487Z] 2022-04-13T06:26:04+02:00:     freebsd12: [203/233] Compiling C object lib/librte_security.a.p/security_rte_security.c.o

[2022-04-13T04:26:04.487Z] 2022-04-13T06:26:04+02:00:     freebsd12: [204/233] Compiling C object drivers/libtmp_rte_mempool_ring.a.p/mempool_ring_rte_mempool_ring.c.o

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00:     freebsd12: [205/233] Generating kernel/freebsd/contigmem with a custom command

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12: gmake[1]: *** [Makefile:167: all] Error 1

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12: gmake: *** [/git_repos/spdk_abi_latest/mk/spdk.subdirs.mk:44: dpdkbuild] Error 2

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00:     freebsd12: machine -> /usr/src/sys/amd64/include

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00:     freebsd12: x86 -> /usr/src/sys/x86/include

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:   04:26:04	-- pkgdep/git@97 -- # trap - ERR

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00:     freebsd12: awk -f /usr/src/sys/tools/makeobjops.awk /usr/src/sys/kern/device_if.m -h

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:   04:26:04	-- pkgdep/git@97 -- # print_backtrace

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:   04:26:04	-- common/autotest_common.sh@991 -- # [[ ehxBET =~ e ]]

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00:     freebsd12: awk -f /usr/src/sys/tools/makeobjops.awk /usr/src/sys/kern/bus_if.m -h

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00:     freebsd12: awk -f /usr/src/sys/tools/makeobjops.awk /usr/src/sys/dev/pci/pci_if.m -h

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:   04:26:04	-- common/autotest_common.sh@993 -- # args=('--disable-tsocks' '--upgrade' '--install-deps')

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:   04:26:04	-- common/autotest_common.sh@993 -- # local args

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:   04:26:04	-- common/autotest_common.sh@995 -- # xtrace_disable

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:   04:26:04	-- common/autotest_common.sh@14 -- # '[' '' '!=' yes ']'

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:   04:26:04	-- common/autotest_common.sh@15 -- # PREV_BASH_OPTS=ehxBET

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:   04:26:04	-- common/autotest_common.sh@16 -- # [[ ehxBET == *\x* ]]

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00:     freebsd12: clang  -O2 -pipe -include rte_config.h  -fno-strict-aliasing -Werror -D_KERNEL -DKLD_MODULE -nostdinc  -I/git_repos/spdk_abi_latest/dpdk/build-tmp -I/git_repos/spdk_abi_latest/dpdk/config -I. -I/usr/src/sys -I/usr/src/sys/contrib/ck/include -fno-common  -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -fdebug-prefix-map=./machine=/usr/src/sys/amd64/include -fdebug-prefix-map=./x86=/usr/src/sys/x86/include   -MD  -MF.depend.contigmem.o -MTcontigmem.o -mcmodel=kernel -mno-red-zone -mno-mmx -mno-sse -msoft-float  -fno-asynchronous-unwind-tables -ffreestanding -fwrapv -fstack-protector -Wall -Wredundant-decls -Wnested-externs -Wstrict-prototypes -Wmissing-prototypes -Wpointer-arith -Wcast-qual -Wundef -Wno-pointer-sign -D__printf__=__freebsd_kprintf__ -Wmissing-include-dirs -fdiagnostics-show-option -Wno-unknown-pragmas -Wno-error=tautological-compare -Wno-error=empty-body -Wno-error=parentheses-equality -Wno-error=unused-function -Wno-error=pointer-sign -Wno-error=shift-negative-value -Wno-address-of-packed-member -Wno-format-zero-length  -mno-aes -mno-avx  -std=iso9899:1999 -c /git_repos/spdk_abi_latest/dpdk/kernel/freebsd/contigmem/contigmem.c -o contigmem.o

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00:     freebsd12: ld -m elf_x86_64_fbsd -d -warn-common --build-id=sha1 -r -d -o contigmem.ko contigmem.o

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:   04:26:04	-- common/autotest_common.sh@17 -- # XTRACE_DISABLED=yes

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00:     freebsd12: :> export_syms

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00:     freebsd12: awk -f /usr/src/sys/conf/kmod_syms.awk contigmem.ko  export_syms | xargs -J% objcopy % contigmem.ko

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00:     freebsd12: objcopy --strip-debug contigmem.ko

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:   04:26:04	-- common/autotest_common.sh@19 -- # set +x

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00:     freebsd12: ninja: build stopped: subcommand failed.

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12: ========== Backtrace start: ==========

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12: in /git_repos/spdk_repo/spdk/test/common/config/pkgdep/git:97 -> install_refspdk(["--install-deps"],["--upgrade"],["--disable-tsocks"])

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:      ...

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    92  			if [[ $OSID != freebsd ]]; then

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    93  				$MAKE -C $output_dir $MAKEFLAGS include/spdk/config.h

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    94  				CONFIG_OCF_PATH="$output_dir/ocf" $MAKE -C $output_dir/lib/env_ocf $MAKEFLAGS exportlib O=$output_dir/ocf.a

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    95  				$output_dir/configure $config_params --with-ocf=$output_dir/ocf.a --with-shared --without-fio

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    96  			fi

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12: => 97  			$MAKE -C $output_dir $MAKEFLAGS

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    98  		)

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    99  	}

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00:     freebsd12: gmake: Leaving directory '/git_repos/spdk_abi_latest'

[2022-04-13T04:26:04.746Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    100

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    101 	function install_qat() {

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    102 		# Dissect the kernel version into maj, min, release and local version

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:      ...

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12: in /git_repos/spdk_repo/spdk/test/common/config/pkgdep/git:551 -> install_sources([])

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:      ...

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    546 		done

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    547

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    548 		if [[ $INSTALL_REFSPDK == true ]]; then

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    549 			# Serialize builds as refspdk depends on spdk

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    550 			[[ $INSTALL_SPDK != true ]] && install_spdk

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12: => 551 			install_refspdk latest

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    552 			install_refspdk LTS

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    553 		fi

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    554 	}

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    555

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    556 	GIT_VERSION=2.25.1

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:      ...

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12: in /git_repos/spdk_repo/spdk/test/common/config/vm_setup.sh:162 -> main([])

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:      ...

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    157 		pre_install

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    158 		install "${packages[@]}"

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    159 	fi

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    160

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    161 	source "$vmsetupdir/pkgdep/git"

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12: => 162 	install_sources

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    163

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    164 	# create autorun-spdk.conf in home folder. This is sourced by the autotest_common.sh file.

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    165 	# By setting any one of the values below to 0, you can skip that specific test. If you are

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    166 	# using your autotest platform to do sanity checks before uploading to the build pool, it is

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:    167 	# probably best to only run the tests that you believe your changes have modified along with

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:      ...

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12: ========== Backtrace end ==========

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:   04:26:04	-- common/autotest_common.sh@1031 -- # xtrace_enable

[2022-04-13T04:26:04.747Z] 2022-04-13T06:26:04+02:00: ==> freebsd12:   04:26:04	-- common/autotest_common.sh@1032 -- # return 0
```

Tested with meson version `0.60.3` and works as expected.

**Comment by kgodzwox** [comment]
The problem occurs with DPDK compiling using meson in version 0.62.0 during image building.
Used version 0.60.3 to solve the issue
https://review.spdk.io/gerrit/c/spdk/spdk/+/12246

**Comment by mikeBashStuff** [comment]
Interesting. From the trace it looks like `ar` is the actual culprit here as that's the part which raises the actual exception. Assuming `ar`'s complaint is valid, it may be that the building of the *.a libs is somehow broken under newer meson? Not sure why would that be case though (also wondering if this may affect other dists as well (like fedora which I think is definitely using newer meson).

**Comment by karlatec** [comment]
@tomzawadzki @mikeBashStuff  you guys think creating a bug in DPDK bug tracker is in order?

**Comment by mikeBashStuff** [comment]
@karlatec I think it's a good idea. Especially that the `gen-pmdinfo-cfile.py` (where this is failing) was affected in the past (c78471cd061a0f64db006c538be34536bc66c12a is one example).

**Comment by kgodzwox** [comment]
Issue reported: https://bugs.dpdk.org/show_bug.cgi?id=993

**Comment by kgodzwox** [comment]
The problem with meson should be fixed in the next point-release of 21.11 DPDK - estimated time-release April 2022

**Comment by jimharris** [comment]
This problem has been fixed with patches from @kgodzwox in our DPDK submodule for now.  DPDK has also fixed this upstream, so we'll be able to switch to that fix when we update our DPDK submodules to the latest DPDK release.
